### PR TITLE
IMBU-120 fluent_api tests

### PR DESCRIPTION
**Pull request by oxtopus**

### CODE REVIEW

**Comment by oxtopus** [comment]
Note: this will only pass with https://github.com/numenta/nupic.research/pull/462 and https://github.com/numenta/nupic.research/pull/459 merged into nupic.research master.  Those need to be reviewed, too.

File: imbu/tests/unit/fluent_api_test.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+class TestFluentAPI(unittest.TestCase):
+
+  def setUp(self):
+    loadPath="/Users/amarshall/nta/numenta-apps/imbu/engine"
+    with _tempsyspath(loadPath):
+      import os
+      os.environ["MODEL_CACHE_DIR"] = "/Users/amarshall/nta/numenta-apps/imbu/cache"
+      os.environ["IMBU_LOAD_PATH_PREFIX"] = loadPath
+      api = importlib.import_module("fluent_api")
```
Comment (BoltzmannBrain) [inline imbu/tests/unit/fluent_api_test.py]: Tests fail here w/ `ImportError: No module named fluent_api`

File: imbu/tests/unit/fluent_api_test.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+@contextlib.contextmanager
+def _tempsyspath(path):
+  sys.path.append(path)
+  yield
+  sys.path.remove(path)
+
+
+class TestFluentAPI(unittest.TestCase):
+
+  def setUp(self):
```
Comment (BoltzmannBrain) [inline imbu/tests/unit/fluent_api_test.py]: Is there a better way to do this than use local paths? Can we just use a tempdir?

File: imbu/tests/unit/fluent_api_test.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+import contextlib
+import importlib
+import sys
+import unittest
+import json
+from paste.fixture import TestApp
```
Comment (BoltzmannBrain) [inline imbu/tests/unit/fluent_api_test.py]: Need to add this to requirements.txt

File: imbu/tests/unit/fluent_api_test.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+import contextlib
+import importlib
+import sys
+import unittest
+import json
```
Comment (BoltzmannBrain) [inline imbu/tests/unit/fluent_api_test.py]: We're using simplejson

**Comment by BoltzmannBrain** [comment]
@oxtopus I left some comments. Tests look good, but not passing.

File: imbu/tests/unit/fluent_api_test.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+# See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+import unittest
+import json
```
Comment (BoltzmannBrain) [inline imbu/tests/unit/fluent_api_test.py]: fluent_api uses simplejson

**Comment by BoltzmannBrain** [comment]
I'm still getting `ImportError: No module named fluent_api`

**Comment by oxtopus** [comment]
You will need to "install" imbu w/ setup.py, now.

**Comment by rhyolight** [comment]
Shouldn't it be `fluentapi` given PEP-8 module naming standards? :trollface:

**Comment by oxtopus** [comment]
@rhyolight Yes.  Also, no longer any reason to call it `fluent`, since there's no longer any relation to https://github.com/numenta-archive/nupic.fluent but what evs.

**Comment by BoltzmannBrain** [comment]
Call it `nlpapi`?

File: imbu/requirements.txt [hunk 6]
```diff
@@ -1,14 +1,8 @@
...
-
-# Fluent
-enum34==1.0.4
-pandas==0.17.0
--e git+https://github.com/numenta/cortipy.git#egg=cortipy-0.1.0
+# nupic.research
 -e git+https://github.com/numenta/nupic.research.git#egg=htmresearch-0.0.1
+
+paste
+pytest
```
Comment (BoltzmannBrain) [inline imbu/requirements.txt]: need `==2.5.1` b/c `error: pytest 2.7.3 is installed but pytest==2.5.1 is required by set(['nupic', 'nupic.bindings'])`

**Comment by oxtopus** [comment]
Try again?  https://ci.numenta.com/browse/IMBU-IMBU1-TI-19 (w/ latest changes) is passing.

**Comment by BoltzmannBrain** [comment]
@oxtopus six failures and one skipped

**Comment by BoltzmannBrain** [comment]
:+1:

**Pull request merged**